### PR TITLE
Cancel performance scenarios when coordinator is cancelled

### PR DESCRIPTION
**Pull request by oehme**

### CODE REVIEW

**Comment by oehme** [comment]
And [here's what happened](https://builds.gradle.org/viewType.html?buildTypeId=Gradle_Check_IndividualPerformanceScenarioWorkersLinux&branch_Gradle_Check_Workers=oehme%2Fperformance%2Ftesting%2Fcancellation&tab=buildTypeStatusDiv) when I cancelled the coordinator build.

File: buildSrc/subprojects/performance/src/main/groovy/org/gradle/testing/DistributedPerformanceTest.groovy [hunk 1]
```diff
@@ -256,6 +263,27 @@ class DistributedPerformanceTest extends PerformanceTest {
...
+        String cancelRequest = '<buildCancelRequest comment="Coordinator build was cancelled" readdIntoQueue="false" />'
+        ["builds", "buildQueue"].each { endpoint ->
+            try {
+                client.post(
+                    path: "$endpoint/id:$buildId",
+                    requestContentType: ContentType.XML,
+                    body: cancelRequest
+                )
+            } catch (HttpResponseException e) {
+                println "Could not cancel build $buildId because $e.message"
```
[the remark this answers is not in the document]
Comment (oehme) [reply buildSrc/subprojects/performance/src/main/groovy/org/gradle/testing/DistributedPerformanceTest.groovy]: We can do that, but there will be no report, the client has already disconnected when the build was cancelled. You'll only find it in the daemon log by logging into the agent.

[the remark this answers is not in the document]
Comment (wolfs) [reply buildSrc/subprojects/performance/src/main/groovy/org/gradle/testing/DistributedPerformanceTest.groovy]: I see. I was thinking that we would see the error as a result of the Gradle build with this exception: https://github.com/gradle/gradle/blob/e2e20afdc1751348843d7c33e743be48f139fef3/subprojects/core/src/main/java/org/gradle/initialization/DefaultBuildCancellationToken.java#L81

File: buildSrc/subprojects/performance/src/main/groovy/org/gradle/testing/DistributedPerformanceTest.groovy [hunk 2]
```diff
@@ -256,6 +263,27 @@ class DistributedPerformanceTest extends PerformanceTest {
...
     }
 
+    private void cancel() {
+        scheduledBuilds.each {
+            cancel(it)
+        }
+    }
+
+    private void cancel(String buildId) {
+        String cancelRequest = '<buildCancelRequest comment="Coordinator build was cancelled" readdIntoQueue="false" />'
```
[the remark this answers is not in the document]
Comment (oehme) [reply buildSrc/subprojects/performance/src/main/groovy/org/gradle/testing/DistributedPerformanceTest.groovy]: Yep

File: buildSrc/subprojects/performance/src/main/groovy/org/gradle/testing/DistributedPerformanceTest.groovy [hunk 3]
```diff
@@ -256,6 +263,27 @@ class DistributedPerformanceTest extends PerformanceTest {
...
+        ["builds", "buildQueue"].each { endpoint ->
```
[the remark this answers is not in the document]
Comment (oehme) [reply buildSrc/subprojects/performance/src/main/groovy/org/gradle/testing/DistributedPerformanceTest.groovy]: No, the docs seem to be wrong. You have to check both.

File: buildSrc/subprojects/performance/src/main/groovy/org/gradle/testing/DistributedPerformanceTest.groovy [hunk 4]
```diff
@@ -183,10 +186,10 @@ class DistributedPerformanceTest extends PerformanceTest {
             requestContentType: ContentType.XML,
             body: buildRequest
         )
-        if (!response.success) {
```
Comment (wolfs) [inline buildSrc/subprojects/performance/src/main/groovy/org/gradle/testing/DistributedPerformanceTest.groovy]: Why did you remove this? Does client.post already throw an exception when the build couldn't be scheduled?

Comment (oehme) [reply buildSrc/subprojects/performance/src/main/groovy/org/gradle/testing/DistributedPerformanceTest.groovy]: It does
